### PR TITLE
Add optional arbitrary impl

### LINKED ISSUE #514: impl Arbitrary for Uuid to enable easy integration of cargo-fuzz
**Is your feature request related to a problem? Please describe.**
It's common to have _obejcts_ that use `Uuid`. And whenever someone wants to use `rust-fuzz`, all the components
of the _object_ need to implement/derive the [Arbitrary trait](https://docs.rs/arbitrary/1.0.0/arbitrary/trait.Arbitrary.html)

**Describe the solution you'd like**
Having an implementation for `Arbitrary` for `Uuid`

Something like this would be a starting point:

```rust
impl Arbitrary<'_> for Uuid {
    fn arbitrary(u: &mut Unstructured<'_>) -> ArbitraryResult<Self> {
        let b = u.bytes(16)?;
        Ok(Uuid::from_bytes(b.try_into().unwrap()))
    }
}
```

However, I assume it will panic most of the time. So, I'm not sure what's the best way to go about it.

### PR DESCRIPTION
**Pull request by KodrAus**
Closes #514 

Adds an optional implementation of the stable `Arbitrary` trait to `Uuid` so it can be used for fuzzing.

We always give you a valid V4 UUID from the input, unless there aren't enough bytes.

I've also streamlined some of our templates a bit, since I find myself removing most of the scaffolding when writing PRs and issues.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Thank you for your interest in contributing to the Uuid Project!
 * [Helpful Links](#helpful-links)
 
 For any questions, please make a post on [users.rust-lang.org][u-r-l-o], post
-on [uuid-rs mailing list] or join our [gitter] channel.
+on [uuid-rs mailing list].
```
Comment (kinggoesgaming) [inline CONTRIBUTING.md:16]: I think we aren't using the mailing list anymore.

**Review by kinggoesgaming — CHANGES_REQUESTED** [review]
Other than removing the mailing list (since you touched that file), everything looks good

File: CONTRIBUTING.md (hunk 1 above)
Comment (kinggoesgaming) [reply CONTRIBUTING.md:16]: I think instead we can unit GitHub Discussions at some point in the future.

Comment (KodrAus) [reply CONTRIBUTING.md:16]: I've used Discussions elsewhere and they've been good for keeping issues focused on actionable things. I'd be keen to use them here in `uuid` too

**Review by kinggoesgaming — APPROVED** [review]

**Comment by KodrAus** [comment]
Thanks for the review @kinggoesgaming! I've been hammering at these last mile stabilization tasks so have made a huge pile of basically unreviewable changes. I was going to open a PR that prepares for our `1.0.0-alpha` release and lists everything out so we can review the whole end result.

**Comment by kinggoesgaming** [comment]
Should be good!

**Pull request merged**